custom_components/ha_hatch/light.py

```diff
@@ -41,15 +43,18 @@ class HatchLightEntity(RestEntity, LightEntity):
     _attr_color_mode = ColorMode.RGB
     _attr_supported_color_modes = {ColorMode.RGB}
 
-    def __init__(self, rest_device: RestPlus, config_turn_on_light: bool):
+    def __init__(self, rest_device: RestIot | RestPlus, config_turn_on_light: bool):
         super().__init__(rest_device, "Light")
         self.config_turn_on_light = config_turn_on_light
 
     def _update_local_state(self):
         if self.platform is None:
             return
         _LOGGER.debug(f"updating state:{self.rest_device}")
-        self._attr_is_on = self.rest_device.is_on
```

Comment:
this looks like an unintentional change to the old behavior, the original code should be in this else but instead is in the if

Reply:
Moved the new gen 2 light to its own object so this no longer applies.